src/controllers/entry.controller.ts

```diff
@@ -163,8 +172,55 @@ export class GoodEntryController {
       },
       description: '',
     })
-    entry: Omit<GoodEntry, 'id'>,
+    entry: Partial<GoodEntry>,
   ): Promise<unknown> {
-    return this.goodEntryRepository.create(entry);
+    console.log('Data received:', entry);
+    const pogProfileParams: PogProfileParams = {
+      userId: '',
+      email: '',
+      doGooder: '',
+    };
+
+    if (entry.userId) {
+      Object.assign(pogProfileParams, {userId: entry.userId});
+    }
+
+    if (entry.email) {
+      Object.assign(pogProfileParams, {email: entry.email});
+    }
+
+    if (entry.doGooder) {
+      Object.assign(pogProfileParams, {doGooder: entry.doGooder});
+    }
+
+    const pogProfile = await this.pogProfileService.findOrCreatePogProfile(
+      pogProfileParams,
+    );
+
+    const goodEntry = new GoodEntry(entry);
+    Object.assign(goodEntry, {
```

Comment:
I noticed several `Object.assign()`'s throughout the code; why not use assignment through property accessors? i.e. `goodEntry.prop = value;`

Reply:
if it's for only one field, yes. Object.assign become interesting when you more than one field to update.

Reply:
interesting indeed, thank you for explaining. is there any benefit to using `Object.assign()` over spread syntax, both in general and in the context of this application?
`obj = {...obj, prop1:val1, prop2:val2};`

Reply:
in practice, they are the same and if you're using the object as a DTO to pass data, then it's equivalent. So here both syntax can be used as we're not having fancy classes with complex behaviours.

 The spread syntax creates a totally new object and doesn't leverage the built-in behaviour of the object (setter, getter, identity) that an object.assign() will. 
 If I'm object.assigning a GoodEntry, it stays a GoodEntry.
 if I'm using a spread, I lose the type.